File: java/gov/nasa/gsfc/gmsec/api5/MessageSpecification.java
/*
 * Copyright 2007-2024 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */

/**
 * @file MessageSpecification.java
 */

package gov.nasa.gsfc.gmsec.api5;

import java.util.ArrayList;
import java.util.Collection;


/**
 * The MessageSpecification class represents a message defintion derived
 * from a message schema.
 *  
 * @see FieldSpecification
 * @see Specification
 * @see MessageFactory#getSpecification()
 */
public class MessageSpecification
{
	private String                        m_schemaID;
	private String                        m_subjectTemplate;
	private ArrayList<FieldSpecification> m_fieldSpecs;


	private MessageSpecification()
	{
	}


	private MessageSpecification(String schemaID, String subjectTemplate, ArrayList<FieldSpecification> fieldSpecs)
	{
		m_schemaID        = schemaID;
		m_subjectTemplate = subjectTemplate;
		m_fieldSpecs      = fieldSpecs;
	}


	/**
	 * Returns the schema (template) ID for this Message Specification.
	 *
	 * @return The schema ID for this Message Specification.
	 */
	public String getSchemaID()
	{
		return m_schemaID;
	}


	/**
	 * Returns the subject template for this Message Specification.
	 *
	 * @return The subject template for this Message Specification.
	 */
	public String getSubjectTemplate()
	{
		return m_subjectTemplate;
	}


	/**
	 * Returns the collection of FieldSpecification objects for this Message Specification.
	 *
	 * @return The collection of FieldSpecification objects.
	 */
	public Collection<FieldSpecification> getFieldSpecifications()
	{
		return m_fieldSpecs;
	}
}

File: framework/src/gmsec5/MessageSpecification.cpp
/*
 * Copyright 2007-2025 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */

#include <gmsec5/MessageSpecification.h>

#include <gmsec5/internal/InternalMessageSpecification.h>


using namespace gmsec::api5;
using namespace gmsec::api5::internal;
using namespace gmsec::api5::util;


MessageSpecification::MessageSpecification(const char* schemaID, const char* subjectTemplate, List<FieldSpecification*>& fieldSpecs)
	: m_internal(new InternalMessageSpecification(schemaID, subjectTemplate, fieldSpecs))
{
}


MessageSpecification::~MessageSpecification()
{
	delete m_internal;
}


const char* MessageSpecification::getSchemaID() const
{
	return m_internal->getSchemaID();
}


const char* MessageSpecification::getSubjectTemplate() const
{
	return m_internal->getSubjectTemplate();
}


const List<FieldSpecification*>& MessageSpecification::getFieldSpecifications() const
{
	return m_internal->getFieldSpecifications();
}
